src/web/releases.rs

```diff
@@ -497,6 +497,69 @@ impl Default for Search {
     }
 }
 
+fn redirect_to_random_crate(req: &Request, conn: &mut PoolClient) -> IronResult<Response> {
+    // since there is a chance of this query returning an empty result,
+    // we retry a certain amount of times, and log a warning.
+    for _ in 0..20 {
+        let rows = ctry!(
+            req,
+            conn.query(
+                "WITH params AS (
+                    -- get maximum possible id-value in crates-table
+                    SELECT last_value AS max_id FROM crates_id_seq
+                )
```

Comment:
Don't CTEs inhibit optimizations? What do you think about inlining it to the one place it's used?

Reply:
CTEs can be also used as optimization fence when the optimizer doesn't build the best query plan, often developers know the data better. 

In this specific case it doesn't make a difference, so it's purely for readability (IMHO)